src/gt4py/next/program_processors/runners/dace_iterator/workflow.py

```diff
@@ -121,11 +118,23 @@ class Meta:
 
 class CompiledDaceProgram(stages.CompiledProgram):
     sdfg_program: dace.CompiledSDFG
-    sdfg_scalar_args: dict[str, int]  # map arg to its position in program ABI
+    sdfg_arg_position: list[
+        int
+    ]  # map SDFG argument to its position in program ABI, -1 if argument is not used
+
+    def __init__(self, program):
+        # extract position of arguments in program ABI
+        sdfg_arglist = program.sdfg.signature_arglist(with_types=False)
+        sdfg_arg_pos_mapping = {param: pos for pos, param in enumerate(sdfg_arglist)}
+        sdfg_used_symbols = program.sdfg.used_symbols(all_symbols=False)
 
-    def __init__(self, program, scalar_args):
         self.sdfg_program = program
-        self.sdfg_scalar_args = scalar_args
+        self.sdfg_arg_position = [
+            sdfg_arg_pos_mapping[param]
+            if param in program.sdfg.arrays or param in sdfg_used_symbols
+            else -1
+            for param in program.sdfg.arg_names
```

Comment:
Totally optional:
- Instead of using `-1` to denote invalid (I guess) use `None`, because `-1` could be understood at "at the end".
- And what does `-1` mean anyway, an argument that is neither a symbol or an array?
- You could construct `arg_names` directly in the correct order, see [PR#1374](https://github.com/GridTools/gt4py/pull/1374) for more, this would simplify a lot. As far as I know nothing depends on it, as I was the one who introduced that feature and if you did not use it in the mean time there should be no problems. (But in that case I would add a big fat note for people like me.)